Makefile

```diff
@@ -152,13 +156,12 @@ endif # CONFIG_BUILD_KERNEL
 dirlinks:
 	$(Q) $(MAKE) -C platform dirlinks TOPDIR="$(TOPDIR)" APPDIR="$(APPDIR)"
 
-context_rest: $(foreach SDIR, $(CONFIGURED_APPS), $(SDIR)_context)
+context_all: $(foreach SDIR, $(CONFIGURED_APPS), $(SDIR)_context)
+register_all: $(foreach SDIR, $(CONFIGURED_APPS), $(SDIR)_register)
 
-context_serialize:
-	$(Q) $(MAKE) -C builtin context TOPDIR="$(TOPDIR)" APPDIR="$(APPDIR)"
```

Comment:
aren't you removing the special case of running context on builtin first? I don't see where builtin is called after your change

Reply:
the special case of running context on builtin first, is useless, so I removed.

You can see there is no "context:: " in builtin/Makefile, then it will use common context:: in Application.mk.
That is do the action of register, but builtin folder has no PROGNAME itself.

so we can remove this:
$(Q) $(MAKE) -C builtin context TOPDIR="$(TOPDIR)" APPDIR="$(APPDIR)"

Reply:
I'm unsure we should remove the call to a target such as context for a subdirectory because it is unused. context might someday be required in builtin and it may not be trivial to understand it is not being invoked. These targets are invoked for all subdirectories in NuttX, so let's keep it even while right now it is not used.
From what I understand $(CONFIGURED_APPS) does not contain "builtin", so unless you explicitly call it, it will not be called at all. Just add "builtin" instead the foreach of the context target. That will be enough I think.

Reply:
@v01d $(CONFIGURED_APPS) actually contain "builtin", here is the code from https://github.com/apache/incubator-nuttx-apps/blob/master/builtin/Make.defs#L37:
```
ifeq ($(CONFIG_BUILTIN),y)
CONFIGURED_APPS += $(APPDIR)/builtin
endif
```

the old Makefile build the context target of builtin twice, this patch fix this issue by removing the explicit invocation.

Reply:
Ok then!